share/functions/__fish_whatis_current_token.fish

```diff
@@ -1,11 +1,25 @@
 # This function is typically bound to Alt-W, it is used to list man page entries
 # for the command under the cursor.
-function __fish_whatis_current_token -d "Show man page entries related to the token under the cursor"
-    set -l tok (commandline -pt)
+function __fish_whatis_current_token -d "Show man page entries or function description related to the token under the cursor"
+    set -l token (commandline -pt)
 
-    if test -n "$tok[1]"
+    if test -n "$token"
         printf "\n"
-        whatis $tok[1]
+        whatis $token 2>/dev/null
```

Comment:
I think we can give precedence to functions and builtins over commands by switching on `type -p $tok[1]`, and only run `whatis` for files.

Reply:
Ah yeah; functions can override commands